### PR TITLE
feat: favorites(wishlist), cart functionality,billing, saving wishlis…

### PR DESCRIPTION
**Pull request by percius47**
-favorites
-cart functionality
-billing
-saving wishlist from cart/menu
-favicons, Titles, Cart/Wishlist Item length in Navbar
[Live Site](https://windys.netlify.app/)

### CODE REVIEW

File: ecom-mockman/src/Components/FavCard.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import nonvegIcon from "../images/nonveg.png";
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { useCart } from '../contexts/cart-context';
+import { useAuth } from '../contexts/auth-context';
+import { useWishlist } from '../contexts/wish-context';
+import { toast,Toaster } from "react-hot-toast";
+function FavCard({product}) {
+  const {_id, title, qty, price, image,veg,Cal,description } = product;
```
Comment (Sen-442b) [inline ecom-mockman/src/Components/FavCard.jsx:12]: any specific reason to use uppercasing for `Cal` here?

File: ecom-mockman/src/Components/FavCard.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                  
+                    {
+                    !veg &&
+                    <img class="v-nveg" src={nonvegIcon} alt="nonveg"/>}
+                        <h6 class="card-subtitle">    {Cal} Cal | {description}</h6>
+                      </div>
+                      <button 
+        className={isAuth && itemInCart?'primary-btn':'secondary-btn font-size-md'}
+          onClick={() =>{
+            console.log("add clicked", isAuth,itemInCart,cartState);
```
Comment (Sen-442b) [inline ecom-mockman/src/Components/FavCard.jsx:62]: You can remove redundant debuggers

File: ecom-mockman/src/Components/FavCard.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    toggleWishlist,
+    loading: wishlistLoading,
+  } = useWishlist();
+  const { cartState, addToCartHandler } = useCart();
+  const { isAuth,navigate } = useAuth();
+  const itemInCart = cartState.find((item) => item._id === _id);
+  const itemInWishlist = wishlistState.find((item) => item._id === _id);
+  return (
+    <div>
+         <div class="cart-card">
```
Comment (Sen-442b) [inline ecom-mockman/src/Components/FavCard.jsx:31]: You can replace `class` with react `className` attribute

**Review by someshtiwarii — APPROVED** [review]
clean UI.. good to go

**Comment by Sen-442b** [comment]
From Live  Site:-

1. The `sort by` radio buttons are looking like checkboxes
2. You can add  a  demo credentials for login feature

**Review by Sen-442b — APPROVED** [review]
- Overall code structure looks clean
- Consistent variable naming
- Clean and intuitive UI

**Pull request merged**